zebra/zebra_rib.c

```diff
@@ -2121,6 +2124,8 @@ static void rib_process_result(struct zebra_dplane_ctx *ctx)
 			if (re) {
 				UNSET_FLAG(re->status, ROUTE_ENTRY_FAILED);
 				SET_FLAG(re->status, ROUTE_ENTRY_INSTALLED);
+				/* Increment the installed re count for the nhe */
+				re->nhe->installed_re_count++;
```

Comment:
no no - we really can't reach-around into the nhe/nhg struct from the rib module.

Reply:
There is a PR https://github.com/FRRouting/frr/pull/18722/ which also introduces nhe->re pointer and taking action to reprogram routes if certain nhg failed to install upon interface flap.

Please elaborate your thought process.